indico/modules/events/registration/models/checks.py

```diff
@@ -0,0 +1,184 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2025 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from sqlalchemy.ext.declarative import declared_attr
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy import UTCDateTime
+from indico.core.db.sqlalchemy.custom.int_enum import PyIntEnum
+from indico.util.date_time import now_utc
+from indico.util.enum import RichIntEnum
+from indico.util.i18n import L_
+from indico.util.locators import locator_property
+from indico.util.string import format_repr
+
+
+class CheckType(RichIntEnum):
+    __titles__ = [None, L_('Checked in'), L_('Checked out')]
+    check_in = 1
+    check_out = 2
+
+
+class RegistrationCheckRule(RichIntEnum):
+    __titles__ = [None, L_('Once'), L_('Multiple'), L_('Once daily')]
+
+    once = 1
+    multiple = 2
+    once_daily = 3
+
+
+class RegistrationCheckType(db.Model):
+    """Checks to be performed on registrations."""
+
+    __tablename__ = 'check_types'
+
+    @declared_attr
+    def __table_args__(cls):
+        return (db.Index('ix_uq_check_types_title_lower', cls.event_id, db.func.lower(cls.title), unique=True),
+                db.CheckConstraint('((is_system_defined = true AND event_id IS NULL) '
+                                   'OR (is_system_defined = false AND event_id IS NOT NULL))',
+                                   'valid_is_system_defined'),
+                {'schema': 'event_registration'})
+
+    #: The ID of the object
+    id = db.Column(
+        db.Integer,
+        primary_key=True
+    )
+    #: The ID of the event where this check was created
+    event_id = db.Column(
+        db.Integer,
+        db.ForeignKey('events.events.id'),
+        index=True,
+    )
+    #: The name of the check
+    title = db.Column(
+        db.String,
+        nullable=False
+    )
+    #: The associated `RegistrationCheckRule`
+    rule = db.Column(
+        PyIntEnum(RegistrationCheckRule),
+        default=RegistrationCheckRule.once,
+        nullable=False
+    )
+    #: If this is a system defined check type
+    is_system_defined = db.Column(
```

Comment:
There no way to add system-defined check types from the UI, right? Shouldn't it be possible in the admin area? /cc @ThiefMaster

Reply:
Right, I will be adding that